src/scanomf.d

```diff
@@ -187,7 +187,8 @@ void scanOmfObjModule(void delegate(const(char)* name, int pickAny) pAddSymbol,
                 parseName(&p, name.ptr);
                 p += (recTyp == PUBDEF) ? 2 : 4; // skip offset
                 parseIdx(&p); // skip type index
-                pAddSymbol(name.ptr, 0);
+                const len = strlen(name.ptr);
+                pAddSymbol(name[0 .. len], 0);
```

Comment:
Originally I had it all be the two-step approach, but then the compiler complained about the `goto` skipping the initialization. I had it as two-step because it's simpler to read.

Keep in mind that my purpose with this incremental change is to fix the interface to the function, so that subsequent improvements are encapsulated and much easier to review. I plan to improve the internals later.